hardware/BleBox.cpp

```diff
@@ -586,14 +569,9 @@ namespace http {
 			}
 
 			std::string hwid = request::findValue(&req, "idx");
-			if (hwid == "")
-				return;
-			int iHardwareID = atoi(hwid.c_str());
-			CDomoticzHardwareBase *pBaseHardware = m_mainworker.GetHardware(iHardwareID);
+			CDomoticzHardwareBase *pBaseHardware = GetHardwareBase(hwid, HTYPE_BleBox);
```

Comment:
Maybe then extend the m_mainworker.GetHardware function with an 'Optional' Type parameter then
This code should not be in the webserver class
And it is also not entirely true, as on many places in the code we check for multiple types
But maybe keep the code refactoring for later, and just use the normal method in the blebox class